airflow/www/views.py

```diff
@@ -186,13 +186,19 @@ def get_date_time_num_runs_dag_runs_form_data(www_request, session, dag):
     """Get Execution Data, Base Date & Number of runs from a Request"""
     date_time = www_request.args.get('execution_date')
     if date_time:
-        date_time = timezone.parse(date_time)
+        try:
+            date_time = timezone.parse(date_time)
+        except ParserError:
+            return {"error": "Invalid execution_date"}
```

Comment:
Instead of returning a dict and checking it outside, it’d be better to simply catch and handle `ParserError` outside this function.

Reply:
I did the same initially but the issue was that either of `base_date` and `execution_date` can raise error and the error message cannot be set appropriately in the caller of `get_date_time_num_runs_dag_runs_form_data(www_request, session, dag)`